File: pkg/server/pokete/msg/fight/attack_result.go
package fight

import "github.com/lxgr-linux/pokete/bs_rpc/msg"

const AttackResultType msg.Type = "pokete.fight.attackResult"

type Result uint

const (
	Result_Attack Result = iota
	Result_RunAway
	Result_Item
	Result_ChoosePoke
)

type AttackResult struct {
	msg.BaseMsg
	Result Result `json:"result"`
	Attack string `json:"attack"`
	Item   string `json:"item"`
}

func (r AttackResult) GetType() msg.Type {
	return AttackResultType
}

File: pokete_classes/multiplayer/msg/fight/attack_result.py
from typing import TypedDict

import bs_rpc

ATTACK_RESULT_TYPE = "pokete.fight.attackResult"


class AttackResultData(TypedDict):
    result: int
    attack: str | None
    item: str | None


class AttackResult(bs_rpc.Body):
    def __init__(self, data: AttackResultData):
        super().__init__(ATTACK_RESULT_TYPE, data)
